packages/truffle-deployer/test/deployer.js

```diff
@@ -51,7 +56,10 @@ describe("Deployer (sync)", function() {
       }
     }
     deployer = new Deployer(options);
-    reporter = new Reporter(deployer);
+    reporter = new Reporter();
+    reporter.setDeployer(deployer);
+    reporter.setMigration(mockMigration);
+    reporter.listen();
```

Comment:
Suggestion: what if we used a builder pattern for this?

```javascript
reporter = new Reporter()
  .setDeployer(deployer)
  .setMigration(mockMigration)
  .listen();
```

Reply:
The Reporter is instantiated once for a complete migrations cycle, which allows us to summarize their aggregate result. But it needs to be consume individual deployer and migration instances as they arise, hence the decoupling here. Definitely looks like we could simplify by having an `init` method or something that did all this in one place. 

Possibly make this change at `async-deployer`.